### ISSUE TITLE
java.lang.NoClassDefFoundError generated from Import Spring Getting Started Content

### ISSUE THREAD

**Issue by jonathn6** [spring-projects/spring-ide#261]
I was instructed to make use of the STS (Spring Tool Suite) for a java course I am enrolled in.  I have downloaded STS version 3.9.3 with platform Eclipse Oxygen 3 (4.7.3).

When I start up STS, I do the following:
1. Click File -> New -> Import Spring Getting Started Content
2. I do a search on rest
3.  I select rest services with a code set of initial and complete
4.  Once the application is generated, I navigate to the gs-rest-service-complete/src/main/java/hello subdirectory and I see Application.java, Greeting.java, and GreetingController.java.
5.  I click on the green run button and I get the following output in the console:
Exception in thread "main" java.lang.NoClassDefFoundError: org/springframework/util/Assert
	at org.springframework.boot.SpringApplication.<init>(SpringApplication.java:263)
	at org.springframework.boot.SpringApplication.<init>(SpringApplication.java:247)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1246)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1234)
	at hello.Application.main(Application.java:10)
Caused by: java.lang.ClassNotFoundException: org.springframework.util.Assert
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 5 more

I did not make any modifications to any configuration settings or any code.  Why would I be getting this error?  How do I resolve it?

Thank you.
Jonathan

**Comment by BoykoAlex** [comment]
It seems a lot like you've started STS 3.9.2 instead of 3.9.3. 3.9.2 is known to have a problem with Spring Boot 2.0.

**Comment by jonathn6** [comment]
According to About Spring Tool Suite, I am running version 3.9.3.RELEASE with a build ID of 201803200915 and a Platform of Eclipse Oxygen.3 (4.7.3)

**Comment by BoykoAlex** [comment]
What kind of `Run Configuration` have you created? It should be of type `Spring Boot App` or possibly a `Java Application`. Is that the case?
Also try right-clicking on the project `Maven -> Update Project...`
You should be running the `gs-rest-service-complete` project not the `gs-rest-service-initial`...

**Comment by kdvolder** [comment]
The error is indicative of a problem with the classpath. Since you say you din't make any changes to the project it should really work out of the box. 

It could be something went wrong when the project was being initialized / configured in Eclipse. An issue with maven, the local maven repo, network connection problem... or something like that. This might have left the project classpath in a broken state.

So perhaps you can try to force it to 'repair' itself. Right click on the project and select "Maven >> Update Project". Check the 'update snapshots and releases checkbox' as well.

Then you can also select from "Project" menu the 'clean' command to force project to rebuild.

Now try running it again.

If it still doesn't work we can do something a bit more drastic.

- Delete the project from the workspace
- Close STS
- Delete the `~/.m2/repository` directory from your hard drive.
- Reopen STS and reimport the project and try again.

Not sure it will help. But worth a shot.

**Comment by kdvolder** [comment]
For the record. I just tried importing the `gs-rest-service-complete` project and it runs fine for me, using a fresh STS 3.9.3. So I think some kind of problem that happened during the project import is probably the cause. Though it will be hard to know precisely what went wrong.

**Comment by jonathn6** [comment]
Someone suggested that STS won't work with Java 10.  I uninstalled Java 10 and installed Java 8. But I still experience the same issue.  I will try your other suggestions shortly

Jonathan

**Comment by kdvolder** [comment]
BTW: 'the green button' isn't totally unambiguous. There's quite a number of buttons that might fit that description. So... for the record... the one I clicked to run the project is the little 
![start-button](https://user-images.githubusercontent.com/1002156/38049638-9b2f3c46-327d-11e8-8e55-99c12ca956a7.png) in the 'Boot Dashboard'. 

And yes, running in Java 10 isn't supported with STS 3.9.3. Sticking to Java 8 is probably a better choice, its also what i was using.

After you made changes to the JVM installation on your machine, its also a good idea to check your workspace configuration as it is very likely Eclipse still 'remembers' the JVM you uninstalled as the workspace default JRE. And this will certainly cause all kinds of problems / errors when it tries to use that JVM which no longer exists on your machine.

**Comment by jonathn6** [comment]
Kris:

Just to update you, I ended up removing Java 10 from my system and I installed Java 8.  Then, I deleted the Spring Tool Suite as well.  I also deleted the workspace folder.  Then, I removed 2 versions of eclipse that I had installed on my computer from previous courses that I have taken.  I then rebooted my machine.  After the machine booted, I downloaded STS version 3.9.2 64 bit (which includes a compatible version of eclipse).  I then used the wizard to generate the application, and it raw with no problems.  Thanks for your input and assistance.

Jonathan

Sent from Mail for Windows 10
